File: Coding/AnalisadorLexico/teste.js
class Person {
    constructor(name, age) {
        this.name = name;
        this.age = age;
    }

    // Método para obter o nome
    getName() {
        return this.name;
    }

    // Método para definir o nome
    setName(newName) {
        this.name = newName;
    }

    // Método para obter a idade
    getAge() {
        return this.age;
    }

    // Método para definir a idade
    setAge(newAge) {
        this.age = newAge;
    }

    // Método para apresentar a pessoa
    introduce() {
        console.log(`Olá, meu nome é ${this.name} e eu tenho ${this.age} anos.`);
    }
}

// Exemplo de uso
const person1 = new Person('Alice', 30);
person1.introduce(); // Olá, meu nome é Alice e eu tenho 30 anos.

person1.setName('Bob');
person1.setAge(25);
person1.introduce(); // Olá, meu nome é Bob e eu tenho 25 anos.

console.log(person1.getName()); // Bob
console.log(person1.getAge());  // 25

File: Coding/AnalisadorLexico/teste.py
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    # Método para obter o nome
    def get_name(self):
        return self.name

    # Método para definir o nome
    def set_name(self, new_name):
        self.name = new_name

    # Método para obter a idade
    def get_age(self):
        return self.age

    # Método para definir a idade
    def set_age(self, new_age):
        self.age = new_age

    # Método para apresentar a pessoa
    def introduce(self):
        print(f"Olá, meu nome é {self.name} e eu tenho {self.age} anos.")


# Exemplo de uso
if __name__ == "__main__":
    person1 = Person("Alice", 30)
    person1.introduce()  # Olá, meu nome é Alice e eu tenho 30 anos.

    person1.set_name("Bob")
    person1.set_age(25)
    person1.introduce()  # Olá, meu nome é Bob e eu tenho 25 anos.

    print(person1.get_name())  # Bob
    print(person1.get_age())   # 25
